spec/models/dataset_spec.rb

```diff
@@ -111,10 +111,32 @@
     expect(dataset.last_updated_at).to eq last_updated_at
   end
 
-  describe 'after initializing' do
-    it "generates a unique short_id upon initialising" do
-      dataset = Dataset.new
-      expect(dataset.short_id).to_not be_nil 
+  context 'when creating an imported legacy dataset' do
+    it "generates a unique deterministic short_id based on the legacy UUID" do
+      legacy_id = 'abcdef123456'
+      short_id = Digest::SHA256.hexdigest(legacy_id)[0..6]
+
+      imported_dataset = FactoryGirl.create(:dataset, uuid: legacy_id)
+      expect(imported_dataset.short_id).to eq short_id
+      imported_dataset.destroy
+
+      imported_dataset = FactoryGirl.create(:dataset, uuid: legacy_id)
+      expect(imported_dataset.short_id).to eq short_id
+    end
+
+    it "continues to generate short_ids until it has found a unique one" do
+      legacy_id = '123456abcdef'
+      3.times do
+        FactoryGirl.create(:dataset, uuid: legacy_id)
```

Comment:
How can we have multiple legacy datasets that have the same ID? Should they not be unique?

Reply:
We can't. Was just a quick way to test that the generate_short_id function would loop if there was a clash.

Reply:
Okay. In that case I'm not sure we need the seed. We'd want an exception to be raised if the short_id already exists since there should never be more than one dataset with the same uuid.

Reply:
We skip all the validations on the import so we don't bomb out mid way through. 

I think that's why previously there was the while loop to check if the candidate_short_id is unique, and if not then a new one is generated.